app/grandchallenge/core/templates/challenge.html

```diff
@@ -33,79 +33,78 @@
     {{ block.super }}
 {% endblock %}
 
-{% block sidebar %}
+{% block topbar %}
     {% if challenge %}
-        <div class="col-12 col-md-3 col-lg-2 mb-3">
-            <ul class="nav nav-pills flex-column">
-                {% for page in pages %}
-                    {% if not page.hidden %}
-                        <li class="nav-item">
-                            <a class="nav-link {% if page == currentpage %}active{% endif %}"
-                               href="{{ page.get_absolute_url }}">
-                                {% filter title %}
-                                    {% firstof page.display_title page.title %}
-                                {% endfilter %}
-                            </a>
-                        </li>
-                    {% endif %}
-                {% endfor %}
-
-                {% if challenge.display_forum_link %}
+        <div class="row">
+            <div class="col-md-10">
+                <div class="nav nav-tabs" role="tablist">
                     <li class="nav-item">
-                        <a class="nav-link"
-                           href="{% url 'forum:forum' slug=challenge.forum.slug pk=challenge.forum.pk %}">
-                            Forum
+                        <a class="nav-link {% if request.resolver_match.view_name == 'pages:detail' %} show active{% endif %}"
+                           href="{% url 'pages:detail' challenge_short_name=challenge.short_name page_title=challenge.page_set.first.title %}">
```

Comment:
We need to work out which is the first page as some might be hidden or limited view. In fact, it might be better to use `{% url "pages:home" challenge_short_name=challenge.short_name %}` here, which does this.

Reply:
Yes, I see that `pages:home` returns the detail view with page set as `self.request.challenge.page_set.first()`. But that's no different from what I'm doing up here in the code, right? The home view does not solve the issue if this first page is hidden, or am I missing something?

Reply:
it does catch the case when there is no page at all, which I guess is good, so in any case, it's better to use `pages:home`

Reply:
oh wait, is it the case that because it's querying from the `request` that `self.request.challenge.page_set.first()` will get the first of the set of pages you as the requester / viewer has access to? then, I get how it solves the problem